gateway/src/main/scala/filodb/gateway/conversion/InfluxProtocolParser.scala

```diff
@@ -173,6 +177,15 @@ object InfluxProtocolParser extends StrictLogging {
     }
   }
 
+  // Compares the FIRST key to some bytes
+  def firstKeyEquals(bytes: Array[Byte], offsets: Buffer[Int], compareTo: Array[Byte]): Boolean = {
+    require(offsets.nonEmpty)
+    val firstKeyOffset = keyOffset(offsets(0))
+    val firstKeyLen = valOffset(offsets(0)) - 1 - firstKeyOffset
+    UnsafeUtils.equate(bytes, UnsafeUtils.arayOffset + firstKeyOffset,
```

Comment:
Can this be done without UnsafeUtils? Less unsafe is better.

Reply:
I don't think there's an equivalent here that is as performant.....

Reply:
The buffer has no feature to compare to an array?

Reply:
The only buffer is the Netty one, but by this point it's no longer visible, thus would need a whole bunch of API changes.  I'll look into it separately.